src/daft-plan/src/physical_plan.rs

```diff
@@ -256,6 +256,151 @@ impl PhysicalPlan {
+                Self::FanoutByRange(FanoutByRange { num_partitions, sort_by, .. }) => Self::FanoutByRange(FanoutByRange::new(input.clone(), *num_partitions, sort_by.clone())),
+                Self::ReduceMerge(..) => Self::ReduceMerge(ReduceMerge::new(input.clone())),
+                Self::Aggregate(Aggregate { aggregations, groupby, ..}) => Self::Aggregate(Aggregate::new(input.clone(), aggregations.clone(), groupby.clone())),
+                Self::TabularWriteParquet(TabularWriteParquet { schema, file_info, .. }) => Self::TabularWriteParquet(TabularWriteParquet::new(schema.clone(), file_info.clone(), input.clone())),
+                Self::TabularWriteCsv(TabularWriteCsv { schema, file_info, .. }) => Self::TabularWriteCsv(TabularWriteCsv::new(schema.clone(), file_info.clone(), input.clone())),
+                Self::TabularWriteJson(TabularWriteJson { schema, file_info, .. }) => Self::TabularWriteJson(TabularWriteJson::new(schema.clone(), file_info.clone(), input.clone())),
+                _ => panic!("Physical op {:?} has two inputs, but got one", self),
+            },
+            [input1, input2] => match self {
+                #[cfg(feature = "python")]
+                Self::InMemoryScan(..) => panic!("Source nodes don't have children, with_new_children() should never be called for source ops"),
+                Self::TabularScan(..)
+                | Self::EmptyScan(..)
+                | Self::TabularScanParquet(..)
+                | Self::TabularScanCsv(..)
+                | Self::TabularScanJson(..) => panic!("Source nodes don't have children, with_new_children() should never be called for source ops"),
+                Self::HashJoin(HashJoin { left_on, right_on, join_type, .. }) => Self::HashJoin(HashJoin::new(input1.clone(), input2.clone(), left_on.clone(), right_on.clone(), *join_type)),
+                Self::BroadcastJoin(BroadcastJoin {
+                    left_on,
+                    right_on,
+                    join_type,
+                    is_swapped,
+                    ..
+                }) => Self::BroadcastJoin(BroadcastJoin::new(input1.clone(), input2.clone(), left_on.clone(), right_on.clone(), *join_type, *is_swapped)),
+                Self::SortMergeJoin(SortMergeJoin { left_on, right_on, join_type, num_partitions, left_is_larger, needs_presort, .. }) => Self::SortMergeJoin(SortMergeJoin::new(input1.clone(), input2.clone(), left_on.clone(), right_on.clone(), *join_type, *num_partitions, *left_is_larger, *needs_presort)),
+                Self::Concat(_) => Self::Concat(Concat::new(input1.clone(), input2.clone())),
+                _ => panic!("Physical op {:?} has one input, but got two", self),
+            },
+            _ => panic!("Physical ops should never have more than 2 inputs, but got: {}", children.len())
+        }
+    }
+
+    pub fn name(&self) -> String {
+        format!("{:?}", self)
+    }
+
+    pub fn multiline_display(&self) -> Vec<String> {
```

Comment:
I like `multiline_display`!

I wonder if we could even make this a `MultilineDisplay` trait and extend it so that if all the struct's members are `: MultilineDisplay` then deriving `MultilineDisplay` is trivial 😛

Reply:
(I think this is only nice-to-have, not a blocking comment)